### PR TITLE
Drag and drop tasks

### PR DESCRIPTION
**Pull request by neo-tahi**
This implementation relies solely on the backend to change the position. It uses the `invoke` method in Ember to reload the tasks dragged around to refresh the positions, resulting in some wait time before the changes appear. 

This approach is the least error prone, we tried client-side sorting but found it to be really error-prone.

### CODE REVIEW

File: app/services/paper_factory.rb [hunk 1]
```diff
@@ -19,8 +19,9 @@ def apply_template
     template.phase_templates.each do |phase_template|
       phase = paper.phases.create!(name: phase_template['name'])
 
-      phase_template.task_templates.each do |task_template|
-        create_task(task_template, phase)
+      phase_template.task_templates.each_with_index do |task_template, index|
+        position = index + 1
```
Comment (rizwanreza) [inline app/services/paper_factory.rb]: We shouldn't need this, I will see if I can remove it.

**Comment by mikem** [comment]
Should there be a default ordering somewhere for tasks? Maybe on `Phase`, `has_many :tasks, order: :position` or something like that?

**Comment by rizwanreza** [comment]
@mikem: We're ordering by position in Ember land.

**Comment by mikem** [comment]
Ah yes, there it is, in `PhaseController`. Thanks @rizwanreza!

:+1: looks OK to me.

**Comment by 4thethrillofit** [comment]
Quick note: there is currently a bug with the feature where if you append a card from a different column to column X twice (to the end of the column), we get `index out of bound` and all the card disappear. I started looking into it but have not fixed it yet.

**Pull request merged**